docs/tutorials/course/fairbo.md

```diff
@@ -0,0 +1,506 @@
+
+We can now plot the histograms for:
+
+(1) binary target output for good (1) or bad (0) credit risk;
+
+(2) binary ForeignWorker feature;
+
+(3) binary target output for individuals with ForeignWorker feature equals to "yes";
+
+(4) binary target output for individuals with ForeignWorker feature equals to "no".
+
+
+```python
+fig, (ax1, ax2) = plt.subplots(ncols=2)
+fig.tight_layout(pad=5.0)
+
+sns.countplot(df['target'], ax=ax1)
+ax1.set_title('Target')
+
+sns.countplot(df['ForeignWorker'], ax=ax2)
+ax2.set_title('ForeignWorker')
+plt.show()
+```
+
+We can note that our dataset unbalanced, having more than double positive examples compared to negative ones. The unbalance is ever higher concerning the two subgroups of foreign against local workers.
+
+
+```python
+fig, (ax1, ax2) = plt.subplots(ncols=2)
+fig.tight_layout(pad=5.0)
+
+sns.countplot(df[df["ForeignWorker"] == 'no']['target'], ax=ax1)
+ax1.set_title('Target | Non ForeignWorker')
+
+
+sns.countplot(df[df["ForeignWorker"] == 'yes']['target'], ax=ax2)
+ax2.set_title('Target | ForeignWorker')
+plt.show()
+```
+
+Finally, here we have a possible source of bias: the proportion between positive and negative target label is different between the two subgroups.
+
+
+```python
+bad_credit_risk_foreign = 1.0 - np.array(df[df["ForeignWorker"] == 'yes']['target']).mean()
+bad_credit_risk_local = 1.0 - np.array(df[df["ForeignWorker"] == 'no']['target']).mean()
+
+print(f"Proportion of bad credit risk for foreign workers: {bad_credit_risk_foreign}")
+print(f"Proportion of bad credit risk for non foreign workers: {bad_credit_risk_local}")
+```
+
+## Fairness metrics
+
+We are now interested in measuring the unfairness (bias) of our model $f$ with respect to the sensitive attribute gender (i.e., _is our model discriminating a specific group of people?_). It is important to note that  there is no consensus on a unique definition of fairness of a ML model. In fact, some of the most common definitions are even conflicting.
+
+In this tutorial we decided to select a commonly used measure for bias called Statistical Parity (SP). SP for a binary classification problem requires the trained model to have the same probability of predicting a positive label among the different subgourps $A$ and $B$. Empirically, we are interested in controlling the amount of violation of this constraint, and we define the Difference in Statistical Parity (DSP). Ideally, we would like this quantity to be small than a certain threshold value $\epsilon$:
+
+$$DSP(f) = \Big| \mathbb{P}_{(x,y)} \big[ f(x)>0 \, \big| \, x \text{ in group } A \big] -  \mathbb{P}_{(x,y)} \big[ f(x)>0 \, \big| \, x \text{ in group } B \big] \Big| \leq \epsilon.$$
+
+Another possible definition is called Equal Opportunity (OP), that is defined as the property of having a similar True Poisitve Rate among the different groups. Also in this case we can defined the Difference in Equal Opportunity (DEO), and our goal is to keep it smaller that $\epsilon$:
```

Comment:
would be helpful if "Equal Opportunity" is actually a link to external reference that user can click to learn more about this concept.

Reply:
typo: Poisitve